kratos/includes/parallel_environment.h

```diff
@@ -32,6 +32,23 @@ namespace Kratos
 ///@name Kratos Classes
 ///@{
 
+class KRATOS_API(KRATOS_CORE) EnvironmentManager
+{
+  public:
+    typedef std::unique_ptr<EnvironmentManager> Pointer;
+
+    EnvironmentManager(EnvironmentManager& rOther) = delete;
+
+    virtual ~EnvironmentManager() = default;
+
+    virtual bool IsInitialized() const = 0;
```

Comment:
I don't understand how being an abstract class compiles in this case? It should give some error in `bool ParallelEnvironment::MPIIsInitializedDetail() const`. Isn't it?

Reply:
No I see that it will check it on runtime.

Sorry for the noise!